### PR TITLE
add Anolis support

### PR DESCRIPTION
**Pull request by weidongkl**
## Proposed Commit Message
Add [Anolis](https://openanolis.cn/) OS support

### CODE REVIEW

File: cloudinit/distros/__init__.py [hunk 1]
```diff
@@ -62,6 +62,7 @@
         "photon",
         "rhel",
         "rocky",
+        "anolis",
```
Comment (igalic) [inline cloudinit/distros/__init__.py:65]: this list should be alphabetical

File: cloudinit/net/sysconfig.py [hunk 2]
```diff
@@ -30,6 +30,7 @@
     "openmandriva",
     "rhel",
     "rocky",
+    "anolis",
```
Comment (igalic) [inline cloudinit/net/sysconfig.py:33]: this list should be alphabetical

File: cloudinit/util.py [hunk 3]
```diff
@@ -643,6 +643,7 @@ def _get_variant(info):
             "photon",
             "rhel",
             "rocky",
+            "anolis",
```
Comment (igalic) [inline cloudinit/util.py:646]: this list should be alphabetical

File: config/cloud.cfg.tmpl [hunk 4]
```diff
@@ -4,7 +4,7 @@
 # and base configuration.
 {% set is_bsd = variant in ["dragonfly", "freebsd", "netbsd", "openbsd"] %}
 {% set is_rhel = variant in ["almalinux", "centos", "cloudlinux", "eurolinux",
-                             "miraclelinux", "rhel", "rocky", "virtuozzo" ] %}
+                             "miraclelinux", "rhel", "rocky", "anolis", "virtuozzo" ] %}
```
Comment (igalic) [inline config/cloud.cfg.tmpl:7]: this list should be alphabetical

File: systemd/cloud-init-generator.tmpl [hunk 5]
```diff
@@ -20,7 +20,7 @@ CLOUD_SYSTEM_TARGET="/usr/lib/systemd/system/cloud-init.target"
 {% else %}
 CLOUD_SYSTEM_TARGET="/lib/systemd/system/cloud-init.target"
 {% endif %}
-{% if variant in ["almalinux", "centos", "cloudlinux", "eurolinux", "fedora",
+{% if variant in ["almalinux", "centos", "cloudlinux", "eurolinux", "fedora", "anolis",
```
Comment (igalic) [inline systemd/cloud-init-generator.tmpl:23]: this list should be alphabetical

File: systemd/cloud-init.service.tmpl [hunk 6]
```diff
@@ -12,7 +12,7 @@ After=systemd-networkd-wait-online.service
 {% if variant in ["ubuntu", "unknown", "debian"] %}
 After=networking.service
 {% endif %}
-{% if variant in ["almalinux", "centos", "cloudlinux", "eurolinux", "fedora",
+{% if variant in ["almalinux", "centos", "cloudlinux", "eurolinux", "fedora", "anolis",
```
Comment (igalic) [inline systemd/cloud-init.service.tmpl:15]: this list should be alphabetical

File: tests/unittests/test_cli.py [hunk 7]
```diff
@@ -250,7 +250,7 @@ def test_wb_schema_subcommand_parser(self, capsys):
                     "cloudlinux, cos, debian, eurolinux, fedora, freebsd, "
                     "mariner, miraclelinux, "
                     "openbsd, openEuler, openmandriva, "
-                    "opensuse, photon, rhel, rocky, sles, ubuntu, virtuozzo",
+                    "opensuse, photon, rhel, rocky, anolis, sles, ubuntu, virtuozzo",
```
Comment (igalic) [inline tests/unittests/test_cli.py:253]: this list should be alphabetical

File: tests/unittests/test_util.py [hunk 8]
```diff
@@ -279,6 +293,7 @@
 REDHAT_RELEASE_EUROLINUX_8 = "EuroLinux release 8.4 (Vaduz)"
 REDHAT_RELEASE_MIRACLELINUX_8 = "MIRACLE LINUX release 8.4 (Peony)"
 REDHAT_RELEASE_ROCKY_8 = "Rocky Linux release 8.3 (Green Obsidian)"
+REDHAT_RELEASE_ANOLIS_8 = "Anolis OS 8.2"
```
Comment (igalic) [inline tests/unittests/test_util.py:296]: this list should be alphabetical

File: tests/unittests/test_util.py [hunk 9]
```diff
@@ -1247,6 +1278,7 @@ class TestGetVariant:
             ({"system": "linux", "dist": ("photon",)}, "photon"),
             ({"system": "linux", "dist": ("rhel",)}, "rhel"),
             ({"system": "linux", "dist": ("rocky",)}, "rocky"),
+            ({"system": "linux", "dist": ("anolis",)}, "anolis"),
```
Comment (igalic) [inline tests/unittests/test_util.py:1281]: this list should be alphabetical

File: tools/read-dependencies [hunk 10]
```diff
@@ -26,6 +26,7 @@ DISTRO_PKG_TYPE_MAP = {
     'eurolinux': 'redhat',
     'miraclelinux': 'redhat',
     'rocky': 'redhat',
+    'anolis': 'redhat',
```
Comment (igalic) [inline tools/read-dependencies:29]: this list should be alphabetical

File: tools/render-cloudcfg [hunk 11]
```diff
@@ -31,6 +31,7 @@ def main():
         "rhel",
         "suse",
         "rocky",
+        "anolis",
```
Comment (igalic) [inline tools/render-cloudcfg:34]: this list should be alphabetical

**Review by igalic — CHANGES_REQUESTED** [review]
these lists should be alphabetical

(i might have missed some)

**Review by TheRealFalcon — COMMENTED** [review]
Thanks for the contribution. Do you have any logs/examples of cloud-init running on Anolis with these changes?

**Comment by igalic** [comment]
given that the cloud-init team is on holidays, maybe this bot should take a break too

**Pull request closed without merging**